src/Symfony/Component/HttpFoundation/File/MimeType/FileBinaryMimeTypeGuesser.php

```diff
@@ -43,7 +43,33 @@ public function __construct($cmd = 'file -b --mime %s 2>/dev/null')
      */
     public static function isSupported()
     {
-        return '\\' !== DIRECTORY_SEPARATOR && function_exists('passthru') && function_exists('escapeshellarg');
+        static $supported;
```

Comment:
If you don't mind, here is an equivalent shorter implementation:
```diff
+        static $supported = null;
+
+        if (null !== $supported) {
+            return $supported;
+        }
+
+        if ('\\' === DIRECTORY_SEPARATOR || !function_exists('passthru') || !function_exists('escapeshellarg')) {
+            return $supported = false;
+        }
+
+        ob_start();
+        passthru('command -v file', $return);
+        $binPath = trim(ob_get_clean());
+
+        return $supported = 0 === $return && preg_match('#^/[^\0/]#', $binPath);
```

Reply:
But not as readable...

Reply:
Less cyclomatic complexity so easier to understand actually to me...

Reply:
Cyclomatic complexity might not be a good measure for readability: https://web.eecs.umich.edu/~weimerw/p/weimer-tse2010-readability-preprint.pdf